features/uploads/db.go

```diff
@@ -74,3 +77,45 @@ func InsertUploadRequests(db *sql.DB, userId uint64, uploadRequests []CreateUplo
 	}
 	return ids, nil
 }
+
+func DeleteUploadRequestsWithS3Objects(db *sql.DB, s3Client *s3.S3, bucketName string, userId uint64, requestIDs []uint64) error {
+	// create a transaction to ensure all deletions are atomic
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	// delete the upload requests from the database in one sql query
+	_, err = tx.Exec("DELETE FROM UploadRequests WHERE user_id = $1 AND id = ANY($2)", userId, requestIDs)
+	if err != nil {
+		return err
+	}
+
+	// also delete associated S3 objects
+	for _, requestID := range requestIDs {
+		// retrieve the object key from the database
+		var objectKey string
+		err := tx.QueryRow("SELECT object_key FROM UploadRequests WHERE id = $1", requestID).Scan(&objectKey)
```

Comment:
Some questions:
1. Will the query return values after we deleted them earlier? Even though we are in a transaction I think the query won't return anything. I am unsure. Please check
2. Query in a loop will result in a lot of requests to database. Can you please use 1 query as you did with delete ?

Reply:
1. query inside the loop will not return any values after the corresponding rows have been deleted earlier in the same transaction, it will not result errors, but it will return an empty result set
2. i totally agree, i will change it to one query, we don't need to pay more and it may be better and efficient